packages/next-upgrade/index.ts

```diff
@@ -0,0 +1,223 @@
+    choices.push({
+      title: 'Latest Release Candidate',
+      value: 'rc',
+      description: `Pre-release version for final testing (${nextPackageJson['rc'].version})`,
+    })
+  }
+  choices.push({
+    title: 'Latest Stable',
+    value: 'latest',
+    description: `Production-ready release (${nextPackageJson['latest'].version})`,
+  })
+
+  const shortcutVersion = process.argv[2]?.replace('@', '')
+  if (['canary', 'rc', 'latest'].includes(shortcutVersion)) {
+    // Shortcut
+    targetVersionSpecifier = shortcutVersion as VersionSpecifier
+  } else {
+    const initialVersionSpecifierIdx = await processCurrentVersion()
+
+    const response: Response = await prompts({
+      type: 'select',
+      name: 'version',
+      message: 'What Next.js version do you want to upgrade to?',
+      choices: choices,
+      initial: initialVersionSpecifierIdx,
+    })
+
+    if (!response.version) {
+      return
+    }
+
+    targetVersionSpecifier = response.version
+  }
+
+  // TODO(LichuAcu): support Turborepo
+  const appPackageJsonPath = path.resolve(process.cwd(), 'package.json')
+  let appPackageJson = JSON.parse(fs.readFileSync(appPackageJsonPath, 'utf8'))
+
+  const targetNextPackageJson = nextPackageJson[targetVersionSpecifier]
+
+  if (targetNextPackageJson.version) {
+    if (compareVersions(targetNextPackageJson.version, '15.0.0-canary') >= 0) {
+      await suggestTurbopack(appPackageJson)
+    }
+  }
+
+  fs.writeFileSync(appPackageJsonPath, JSON.stringify(appPackageJson, null, 2))
+
+  // TODO(LichuAcu): keep the original styling of the file
+  console.log('\nUpdating depencies...\n')
+
+  let packageManager: PackageManager = getPackageManager(appPackageJson)
+  const dependencies = `react@${targetNextPackageJson.peerDependencies['react']} react-dom@${targetNextPackageJson.peerDependencies['react-dom']} next@${targetVersionSpecifier}`
+
+  let updateCommand
+  switch (packageManager) {
+    case 'pnpm':
+      updateCommand = `pnpm update ${dependencies}`
+      break
+    case 'npm':
+      updateCommand = `npm install ${dependencies}`
+      break
+    case 'yarn':
+      updateCommand = `yarn add ${dependencies}`
+      break
+    default:
+      updateCommand = `pnpm install next@${targetVersionSpecifier}`
+      break
+  }
+
+  execSync(updateCommand, {
+    stdio: 'inherit',
+  })
+
+  appPackageJson = JSON.parse(fs.readFileSync(appPackageJsonPath, 'utf8'))
+  appPackageJson.dependencies['next'] = targetVersionSpecifier
+  fs.writeFileSync(appPackageJsonPath, JSON.stringify(appPackageJson, null, 2))
+
+  console.log(
+    '\nYour Next.js project has been upgraded successfully. Time to ship! 🚢'
+  )
+}
+
+/*
+ * Logs the current version and returns the index of the current version specifier
+ * in the array ['canary', 'rc', 'latest']
+ */
+async function processCurrentVersion(): Promise<number> {
+  // TODO(LichuAcu): support Turborepo
+  const appLockFilePath = path.resolve(process.cwd(), 'pnpm-lock.yaml')
```

Comment:
Instead of trying to handle package-manager specific lock files, we should probably first check for a `node_modules/next/package.json` file, as that tells us what's *actually* installed, and should be package-manager independent.

https://nodejs.org/api/modules.html#requireresolverequest-options

You can probably use `require.resolve("next/package.json", {paths: [process.cwd()]})` to find where `next` is installed and get the `package.json` file.

If that fails, then we may need to fall back to looking for lockfiles, but that probaly shouldn't be the happy-path.